scvi/train/_trainingplans.py

```diff
@@ -677,6 +695,19 @@ def optimizer_step(self, *args, **kwargs):
     def backward(self, *args, **kwargs):
         pass
 
+    @property
+    def kl_weight(self):
+        """Scaling factor on KL divergence during training."""
+        epoch_criterion = self.n_epochs_kl_warmup is not None
+        step_criterion = self.n_steps_kl_warmup is not None
+        if epoch_criterion:
+            kl_weight = min(1.0, self.current_epoch / self.n_epochs_kl_warmup)
+        elif step_criterion:
+            kl_weight = min(1.0, self.global_step / self.n_steps_kl_warmup)
+        else:
+            kl_weight = 1.0
+        return max(kl_weight, 1e-3)
+
```

Comment:
maybe we write a general util in the file and call that so we don't duplicate code?